pom.xml

```diff
@@ -609,80 +569,125 @@
                 </plugins>
             </reporting>
         </profile>
+
         <profile>
-            <id>java9</id>
-            <activation>
-                <jdk>[1.9,12)</jdk>
-            </activation>
-            <properties>
-                <!-- JDK 9 minimal source and target versions are 1.6 -->
-                <!-- We need 1.8 to use annotations -->
-                <jdkVersion>1.8</jdkVersion>
-            </properties>
-            <build>
-                <plugins>
-                    <plugin>
-                        <artifactId>maven-javadoc-plugin</artifactId>
-                        <configuration>
-                            <source>1.6</source>
-                        </configuration>
-                    </plugin>
-                </plugins>
-            </build>
-            <reporting>
-                <plugins>
-                    <plugin>
-                        <artifactId>maven-javadoc-plugin</artifactId>
-                        <configuration>
-                            <source>1.6</source>
-                        </configuration>
-                    </plugin>
-                </plugins>
-            </reporting>
+        <!-- This is the base profile for running the Checker Framework.  It will always be run,
+             assuming the Java JDK is > 1.7.  (Which is must be for the Checker to work.) -->
+          <id>checkerframework</id>
+          <activation>
+            <jdk>[1.8,)</jdk>
+          </activation>
+          <build>
+            <plugins>
+              <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
```

Comment:
Was this moved here fram line 235?  Given the big diffs, any upstream changes to it will cause a merge conflict.
Would it be possible to just add extra arguments there and have smaller diffs?  I'm OK with every compilation always running the Checker Framework.  I'm also OK with not supporting compilation under Java 8 if it is too difficult (especially since there isn't yet a profile just for Java 8 and lower).

If this is not feasible or would be a huge amount of work, we can forego it, since there are unlikely to be many new changes, but the smaller diffs will make the changes easier to understand as well.

Reply:
was able to clean the diff up with by using <?ignore ... >?